src/components/Home/ClusterDashboardItem.js

```diff
@@ -201,7 +201,8 @@ function ClusterDashboardItem({
             <NameWrapper>{cluster.name}</NameWrapper>
           </TitleWrapper>
           <DeleteDateWrapper>
-            Deleted {relativeDate(cluster.delete_date)}
+            Deleted{' '}
+            {!isNaN(cluster.delete_date) && relativeDate(cluster.delete_date)}
```

Comment:
Does this solve the issue? If the date was `NaN` (or a nullish value), it would be caught by `relativeDate()`.
https://github.com/giantswarm/happa/blob/ed00b918fcbc21cdeb7164f1e0e9729358268c42/src/lib/helpers.tsx#L207

Reply:
It does somehow fix it. The actual object that is coming through is "Invalid Date" when I log `cluster.delete_date`

I think we should investigate why this happens.. it's only on Safari. 

The dates in the API response for that cluster looks like

```
    "create_date": "2020-11-27T13:12:08Z",
    "delete_date": "2021-02-10T16:39:27Z",
```

Which look fine to me. I'm also kind of confused at what point this string becomes a date object. Probably the api client is doing this.

But for now this does fix the immediate crash.

Reply:
Also, if the date is present in the response, it is a `string`, so `isNaN()` will always be `true`.

![image](https://user-images.githubusercontent.com/13508038/107545106-2fd5b580-6bcb-11eb-9e12-266b9887e8e8.png)

Reply:
https://stackoverflow.com/questions/1353684/detecting-an-invalid-date-date-instance-in-javascript

Reply:
To me the fact that it becomes a `Date` object is a bug, and we should solve that instead

Reply:
I agree.. can I make it an issue and call this a temporary workaround

Reply:
Issue to fix this properly is here: https://github.com/giantswarm/giantswarm/issues/15315